### ISSUE TITLE
[popover] moving keyboard focus out of a popover should dismiss it

### ISSUE THREAD

**Issue by brucelawson** [openui/open-ui#1047]
Currently, clicking outside a popover dismisses it. However, using the keyboard to tab through links in a popover, and then to the next link outside the popover does not dismiss it. This means that if the next focussable element is 'underneath' the still-exposed popover, it is not visible. 

This doesn't fall foul of WCAG SC 2.4.11 because "If the user can reveal the focused component without advancing the keyboard focus, the component with focus is not considered visually hidden due to author-created content.", and esc will light dismiss the popover. However, it feels annoying and inconsistent.

~~Should a popover light-dismiss when keyboard focus leaves it? I think so, if it does when  mouse pointer leaves it.~~

Should we create a new declarative way to achieve light dismiss when focus leaves a popover?

**Comment by lukewarlow** [comment]
See https://github.com/openui/open-ui/issues/415 for prior discussions.

Going to update the description a little to make this a discussion about an opt-in way to achieve this declaratively.

**Comment by scottaohara** [comment]
the tldr; of it is that there are types of popovers which should automatically light dismiss when keyboard focus leaves (a custom listbox popup, or a sub-navigation list), and then there are those that shouldn't (non-modal dialog chat windows, some instances of disclosure widget / tip-like popups that describe UI and users may need to refer to it as they type).  it's far easier for developers to add light dismiss behavior to popovers than it is for them to remove that behavior if it was baked in.  

we absolutely should add a declarative way to allow popovers to light dismiss when focus leaves them. and just make sure that when added, it's very clear to authors when to use this feature vs not.

**Comment by brucelawson** [comment]
> disclosure widget / tip-like popups that describe UI and users may need to refer to it as they type

I hadn't thought of that; makes total sense. Cheers Scott.

**Comment by gregwhitworth** [comment]
Closing based on @brucelawson _seeming_ content with the prior resolution. @brucelawson please re-open if you disagree :)

**Comment by lukewarlow** [comment]
This issue is still useful to discuss an opt in mechanism for this.

**Comment by lukewarlow** [comment]
@josepharhar the answer to the question you raised should be above. I think it's that it can't be the default because it's not always the behaviour you'd want. I'm many cases it is needed and I believe we should have and discuss a declarative solution to this.

**Comment by josepharhar** [comment]
I kind of feel like if you're using popover=auto and you can therefore light dismiss with mouse then it means you should also light dismiss with keyboard navigating out of the popover, but I guess that adding a new opt in would also make it easier to ship since there would be no risk of breakage.

**Comment by scottaohara** [comment]
for further context - light dismiss on focus out was explicitly callout out as being a potential problem when going over that feature with Aaron Leventhal, was mentioned as an accessibility concern when discussing with people from Vispero (makers of JAWS screen reader), and when talking about this with coworkers of mine who use screen readers.

The problem is that _some_ types of popovers (e.g, a select's listbox popup and other form control 'pickers') do dismisses on focus leaving - but because popover can be applied to anything, all the other stuff it would be unexpected behavior and could have resulted in people constantly dismissing the content they were attempting to interact with, because tab key just let them leave the popover.   In other words, we did not add auto dismiss to popovers on focus out _specifically for accessibility_.

Additionally when mason, aaron and i were going through this, we generally agreed that it would be far easier for developers to add the autodismiss functionality with JS (it is rather trivial considering all the things popover has done for the developer, by default), than it would be for developers to suppress the behavior.
